hooks/quest/useUpdateToken.ts

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react'
+import { useContractWrite } from 'wagmi'
+import kamonNFTContract from '@/utils/abis/kamonNFT.json'
+
+const useUpdateOwnNFT = (
+  contract: string,
+  tokenId: BigInt,
+  finalTokenUri: string,
+) => {
+
+  const {
+    data: updateData,
+    isError,
+    writeAsync: update
+  } = useContractWrite(
+    {
+      addressOrName: contract,
+      contractInterface: kamonNFTContract.abi
+    },
+    'updateOwnNFT',
+    {
+      args: [tokenId, finalTokenUri],
+    },
+  )
+
+  return {
+    updateData,
+    isError,
+    update,
+  }
+}
+
+export const useUpdateToken = (
```

Comment:
do we need this I think useUpdateOwnNFT is enough.
isSubmitting would be done by using `isLoading` of `useContractWrite`

Reply:
Done in https://github.com/henkaku-center/omise-interface/pull/80/commits/cccee0122a2ab3fd2f730acb61fa20263acf3bfa

But as of now the toast messages related to `updateOwnNFT` are lost.

The `data` returned by `useContractWrite` doesn't help when the user rejects the txn, for example. We get an uncaught exception.

What do you think we should do about this?